src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Caching.cs

```diff
@@ -591,13 +593,13 @@ static void AddListHashCode<TValue>(ref HashCode hc, ConfigurationList<TValue>?
 
                 static void AddHashCode<TValue>(ref HashCode hc, TValue? value)
                 {
-                    if (typeof(TValue).IsValueType)
+                    if (typeof(TValue).IsValueType || !typeof(TValue).IsSealed)
```

Comment:
`IsValue` implies `IsSealed` so the first check is redundant.
```suggestion
                    if (!typeof(TValue).IsSealed)
```

Reply:
That makes `JsonSerializerOptions_ReuseConverterCaches()` fail - I'm not sure exactly why right now as I just installed the preview 3 SDK and now Visual Studio won't let me debug the tests for some reason I've just to fix.

Reply:
My bad, this should have simply been `if (typeof(TValue).IsValue)` without the negation.